fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/scalar/ConstantOperator.java

```diff
@@ -393,7 +393,7 @@ public ConstantOperator castTo(Type desc) throws Exception {
                     throw new SemanticException("Forbidden cast to decimal(precision=0, scale=0)");
                 }
             } catch (AnalysisException e) {
-                throw new SemanticException(e.getMessage());
+                return ConstantOperator.createNull(desc);
```

Comment:
I think returning null is dangerous, FE cast is very weak, some time cast in BE is right but FE can't handle it.

Reply:
so compute in BE?  but doing so would prevent outer join from converting to a narrower outer join or inner join. outer join containing predicates involving decimals would cost more time.